lib/index.js

```diff
@@ -170,16 +170,105 @@ class VideoAsset extends BaseAsset {
 
 exports.VideoAsset = VideoAsset;
 
+function _ensure_cheerio(value) {
+    if (typeof value === 'string')
+        return cheerio.load(value);
+    return value;
+}
+
 class NewsArticle extends BaseAsset {
     constructor() {
         super();
         this._object_type = "ArticleObject";
         this._content_type = "text/html";
     }
 
-    set_document(value) { this._document = cheerio.load(value); }
+    // Takes a Date or something that a Date can be constructed out of
+    set_date_published(value) {
+        this._date_published = value;
+        if (!(value instanceof Date))
+            this._date_published = new Date(value);
+    }
+
+    // Takes a string or array of strings
+    set_authors(value) {
+        this._authors = value;
+        if (!(value instanceof Array))
+            this._authors = [value];
+    }
+
+    set_main_image(asset, caption) {
+        this._main_image = asset;
+        this._main_image_caption = _ensure_cheerio(caption);
+    }
+
     set_section(value) { this._section = value; }
     set_as_static_page() { this._ekn_tags.push(EKN_STATIC_TAG); }
+    set_source(value) { this._source = value; }
+    set_lede(value) { this._lede = _ensure_cheerio(value); }
+    set_body(value) { this._body = _ensure_cheerio(value); }
+    set_read_more_link(value) { this._read_more_link = _ensure_cheerio(value); }
+
+    render() {
+        let main_image_caption = '';
+        if (this._main_image_caption.length)
+            main_image_caption = `<figcaption>${this._main_image_caption.html()}</figcaption>`;
+
+        let main_image = '';
+        if (this._main_image) {
+            main_image = `
+<figure class="main">
+    <img data-libingester-asset-id="${this._main_image.asset_id}">
+    ${main_image_caption}
+</figure>
+            `;
+        }
+
+        this._read_more_link('a').addClass('eos-show-link');
+
+        let body = `
+<header>
+    <div class="context">
+        <p id="context-link"></p>
+    </div>
+    <div class="extra-header-right">
+        <p class="date-published">${this._date_published}</p>
```

Comment:
@ptomato @sgnn7 thinking this could be a good place to use metadata API, because here we want to show the date according to the final user system settings. right?

e.g. 28/12/2017 vs 12/28/2017